core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,149 @@
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.history
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import butterknife.BindView
+import butterknife.ButterKnife
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.drawable
+import org.kiwix.kiwixmobile.core.R.layout
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.R2.id
+import org.kiwix.kiwixmobile.core.extensions.setBitmapFromString
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeFormatter
+
+internal class HistoryAdapter(
+  val historyList: List<HistoryListItem>,
+  val deleteList: List<HistoryListItem>,
+  val itemClickListener: OnItemClickListener
+) : Adapter<ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    return if (viewType == TYPE_ITEM) {
+      Item(inflateLayoutFromParent(parent, layout.item_bookmark_history))
+    } else {
+      Category(inflateLayoutFromParent(parent, layout.header_date))
+    }
+  }
+
+  private fun inflateLayoutFromParent(parent: ViewGroup, layoutId: Int) =
+    LayoutInflater.from(parent.context)
+      .inflate(layoutId, parent, false)
+
+  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+    if (holder is Item) {
+      val history = historyList[position] as HistoryItem
+      setItemDataWithHelpOfHistoryItem(holder, history)
+    } else if (holder is Category) {
+      val date = (historyList[position] as DateItem).dateString
+      setCategoryDataWithHelpOfDate(date, holder)
+    }
+  }
+
+  private fun setCategoryDataWithHelpOfDate(date: String, holder: Category) {
+    val formatter = DateTimeFormatter.ofPattern("d MMM yyyy")
+    val todaysDate = LocalDate.now().format(formatter)
+    val yesterdayDate = LocalDate.now().minusDays(1).format(formatter)
+
+    if (todaysDate != null && todaysDate.contentEquals(date)) {
```

Comment:
this is a candidate for a when block, this null checking pattern has a better form
`todaysDate?.contentEquals(date) == true`

the left hand side evaluates to null if `todaysDate` is null

Reply:
Additionally stop this from comparing strings, you can parse the date and compare date objects